src/components/SortedList.js

```diff
@@ -21,16 +25,20 @@ const SortedList = ({ itemsKeys, data }) => {
     <ul className="sorted-list">
       {itemsKeys
         .sort((a, b) => {
-          // setup the variables with the data we will need
+          // setup the variables with the data we will need.
+          // howSoonA and howSoonB are the numbers of estimated days
+          // until the next purchase. they'll be calcualted every time
+          // the list renders
           let [howSoonA, nameA, howSoonB, nameB] = [
-            data[a].howSoon,
+            Math.abs(getDaysUntilNextPurchase(data[a])),
```

Comment:
Help me understand why you need `Math.abs` here. I’m guessing you found situations where the date when we predicted the item would be bought has passed and this is resulting in a negative value being returned from `getDaysUntilNextPurchase`. Is that right?

If my assumption above is correct, wouldn’t you want to sort on the true value and not the non-negative value? I.e. if the days until next purchase is -14, using `Math.abs` would yield 14 which would put the item in the “kinda soon” group, no?

I’m pretty sure you are doing the right thing and I’m just not following, but thanks for entertaining my question! 🙂

Reply:
Hey, @segdeha! Thanks for this great review. :heart:

Yes, your assumption is correct. 
When an item is out of date, the days until the next purchase will be a negative value. 

However, we only use these values/days `howSoonA` and `howSoonB` when we call `compareItems()`, and we call that function only if both items are out of date or if both items are not. 

When we compare two outdated items both values will be negative. So we use `Math.abs` to make sure that the item with more elapsed days comes second. The item with -14 days until the next purchase should be greater than the one with -10 days.

I think it's safe to use `Math.abs` here because we're always going to compare between two negative values or two positive values. I'll try to add a comment here to make things more clear. Thank you again. :)

Reply:
Ah, OK, that makes sense! Thank you!